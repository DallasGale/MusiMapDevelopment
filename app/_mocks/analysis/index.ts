import SampleThumb from "@assets/sample-thumb.png";

export const summary = {
  items: [
    {
      title: "key",
      value: "d major",
    },
    {
      title: "bpm",
      value: 124,
    },
    {
      title: "instrumental",
      value: "no",
    },
    {
      title: "voice family",
      value: "soprano",
    },
    {
      title: "rhythm presenc",
      value: "39%",
    },
    {
      title: "spectral energy",
      value: "12%",
    },
  ],
};

export const results = {
  data: [
    {
      name: "rock",
      value: 85,
      color: "red",
    },
    {
      name: "pop rock",
      value: 52,
      color: "blue",
    },
    {
      name: "pop",
      value: 50,
      color: "green",
    },
    {
      name: "alternative rock",
      value: 40,
      color: "orange",
    },
  ],
};

export const artist = {
  name: "Kangaroo_MusiQue",
  songTitle: "The_Neverwritten_Role_Playing_Game",
  mp3: "https://commondatastorage.googleapis.com/codeskulptor-demos/DDR_assets/Kangaroo_MusiQue_-_The_Neverwritten_Role_Playing_Game.mp3",
  image: {
    src: SampleThumb.src,
    alt: "Person lying on dirt",
  },
};

export const stats = [
  {
    label: "acoustic attributes",
    value: 9,
  },
  {
    label: "keywords",
    value: 4145,
  },
  {
    label: "genres",
    value: 67,
  },
  {
    label: "situations",
    value: 68,
  },
  {
    label: "acoustic attributes",
    value: 9,
  },
  {
    label: "acoustic attributes",
    value: 9,
  },
];
