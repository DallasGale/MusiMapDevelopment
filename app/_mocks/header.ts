const navItems = {
  primary: [
    {
      label: "Discover",
      link: "/discover",
    },
    {
      label: "Profile",
      link: "/profile",
    },
    {
      label: "Tag",
      link: "/tag",
    },
    {
      label: "Insights",
      link: "/insights",
    },
    {
      label: "Musigraph",
      link: "/musigraph",
    },
  ],
  secondary: [
    {
      label: "Plans",
      link: "/plans",
    },
    {
      label: "About",
      link: "/about",
    },
  ],
};

export default navItems;
