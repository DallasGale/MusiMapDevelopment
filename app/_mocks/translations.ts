export const homePageTranslations = {
  en: {
    headline: "This is what you listen to",
    paragraph:
      "Upload a link or song (under 10MB and 10 minutes) to analyze its mood, genre, use case, BPM, key, and more.",
    upload: {
      placeholder: "Paste a song link (eg. Spotify, Apple Music, YouTube)",
    },
  },
  es: {
    headline: "Esto es lo que escuchas",
    paragraph:
      "Sube un enlace o canción (menos de 10MB y 10 minutos) para analizar su estado de ánimo, género, caso de uso, BPM, clave y más.",
    upload: {
      placeholder:
        "Pega un enlace de canción (ej. Spotify, Apple Music, YouTube)",
    },
  },
  de: {
    headline: "Dit is wat je luistert",
    paragraph:
      "Upload een link of nummer (onder 10MB en 10 minuten) om de stemming, het genre, de gebruikscase, BPM, toonsoort en meer te analyseren.",
    upload: {
      placeholder: "Plak een songlink (bijv. Spotify, Apple Music, YouTube)",
    },
  },
  fr: {
    headline: "C'est ce que vous écoutez",
    paragraph:
      "Téléchargez un lien ou une chanson (moins de 10Mo et 10 minutes) pour analyser son ambiance, son genre, son cas d'usage, ses BPM, sa tonalité et plus encore.",
    upload: {
      placeholder:
        "Collez un lien de chanson (ex. Spotify, Apple Music, YouTube)",
    },
  },
};
