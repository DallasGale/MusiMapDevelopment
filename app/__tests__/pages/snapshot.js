import { render, act } from "@testing-library/react";
import Home from "@app/[lng]/page";
import { MantineProvider } from "@mantine/core";
import theme from "@app/[lng]/theme";
import { headers } from "next/headers";
import { homePageTranslations } from "../../_mocks/translations";

// Mock the headers function
jest.mock("next/headers", () => ({
  headers: jest.fn(),
}));

// Mock your i18n module
jest.mock("../../i18n/index", () => {
  return {
    getT: jest.fn().mockImplementation(async () => {
      // eslint-disable-next-line @typescript-eslint/no-require-imports
      const { headers } = require("next/headers");
      const headerList = await headers();
      const lng = headerList.get("accept-language") || "en";

      // Simulate language change based on headers
      const translations = homePageTranslations[lng] || homePageTranslations.en;

      return {
        t: jest.fn((key, options) => {
          if (key === "text" && options?.returnObjects) {
            return translations;
          }
          return key;
        }),
      };
    }),
  };
});

describe("Home Page Language-Specific Snapshots", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders home page with English translations", async () => {
    // Set English headers
    const mockHeaders = new Map();
    mockHeaders.set("accept-language", "en");
    headers.mockResolvedValue(mockHeaders);

    let component;
    await act(async () => {
      component = await Home();
    });

    const { container } = render(
      <MantineProvider defaultColorScheme="dark" theme={theme}>
        {component}
      </MantineProvider>,
    );

    const h1 = container.querySelector("h1");
    expect(h1).not.toBeNull();
    expect(h1.textContent).toBe(homePageTranslations.en.headline);

    const h2 = container.querySelector("h2");
    expect(h2).not.toBeNull();
    expect(h2.textContent).toBe(homePageTranslations.en.paragraph);

    expect(h1).toMatchSnapshot("home-page-english-h1");
    expect(h2).toMatchSnapshot("home-page-english-h2");
  });

  it("renders home page with Spanish translations", async () => {
    // Set English headers
    const mockHeaders = new Map();
    mockHeaders.set("accept-language", "es");
    headers.mockResolvedValue(mockHeaders);

    let component;
    await act(async () => {
      component = await Home();
    });

    const { container } = render(
      <MantineProvider defaultColorScheme="dark" theme={theme}>
        {component}
      </MantineProvider>,
    );

    const h1 = container.querySelector("h1");
    expect(h1).not.toBeNull();
    expect(h1.textContent).toBe(homePageTranslations.es.headline);

    const h2 = container.querySelector("h2");
    expect(h2).not.toBeNull();
    expect(h2.textContent).toBe(homePageTranslations.es.paragraph);

    expect(h1).toMatchSnapshot("home-page-spanish-h1");
    expect(h2).toMatchSnapshot("home-page-spanish-h2");
  });

  it("renders home page with Duthc translations", async () => {
    // Set English headers
    const mockHeaders = new Map();
    mockHeaders.set("accept-language", "de");
    headers.mockResolvedValue(mockHeaders);

    let component;
    await act(async () => {
      component = await Home();
    });

    const { container } = render(
      <MantineProvider defaultColorScheme="dark" theme={theme}>
        {component}
      </MantineProvider>,
    );

    const h1 = container.querySelector("h1");
    expect(h1).not.toBeNull();
    expect(h1.textContent).toBe(homePageTranslations.de.headline);

    const h2 = container.querySelector("h2");
    expect(h2).not.toBeNull();
    expect(h2.textContent).toBe(homePageTranslations.de.paragraph);

    expect(h1).toMatchSnapshot("home-page-dutch-h1");
    expect(h2).toMatchSnapshot("home-page-dutch-h2");
  });

  it("renders home page with French translations", async () => {
    // Set English headers
    const mockHeaders = new Map();
    mockHeaders.set("accept-language", "fr");
    headers.mockResolvedValue(mockHeaders);

    let component;
    await act(async () => {
      component = await Home();
    });

    const { container } = render(
      <MantineProvider defaultColorScheme="dark" theme={theme}>
        {component}
      </MantineProvider>,
    );

    const h1 = container.querySelector("h1");
    expect(h1).not.toBeNull();
    expect(h1.textContent).toBe(homePageTranslations.fr.headline);

    const h2 = container.querySelector("h2");
    expect(h2).not.toBeNull();
    expect(h2.textContent).toBe(homePageTranslations.fr.paragraph);

    expect(h1).toMatchSnapshot("home-page-french-h1");
    expect(h2).toMatchSnapshot("home-page-french-h2");
  });
});
