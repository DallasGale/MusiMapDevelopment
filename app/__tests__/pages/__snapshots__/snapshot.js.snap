// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Home Page Language-Specific Snapshots renders home page with Duthc translations: home-page-dutch-h1 1`] = `
<h1
  class="headline1"
>
  Dit is wat je luistert
</h1>
`;

exports[`Home Page Language-Specific Snapshots renders home page with Duthc translations: home-page-dutch-h2 1`] = `
<h2
  class="headline2"
>
  Upload een link of nummer (onder 10MB en 10 minuten) om de stemming, het genre, de gebruikscase, BPM, toonsoort en meer te analyseren.
</h2>
`;

exports[`Home Page Language-Specific Snapshots renders home page with English translations: home-page-english-h1 1`] = `
<h1
  class="headline1"
>
  This is what you listen to
</h1>
`;

exports[`Home Page Language-Specific Snapshots renders home page with English translations: home-page-english-h2 1`] = `
<h2
  class="headline2"
>
  Upload a link or song (under 10MB and 10 minutes) to analyze its mood, genre, use case, BPM, key, and more.
</h2>
`;

exports[`Home Page Language-Specific Snapshots renders home page with French translations: home-page-french-h1 1`] = `
<h1
  class="headline1"
>
  C'est ce que vous écoutez
</h1>
`;

exports[`Home Page Language-Specific Snapshots renders home page with French translations: home-page-french-h2 1`] = `
<h2
  class="headline2"
>
  Téléchargez un lien ou une chanson (moins de 10Mo et 10 minutes) pour analyser son ambiance, son genre, son cas d'usage, ses BPM, sa tonalité et plus encore.
</h2>
`;

exports[`Home Page Language-Specific Snapshots renders home page with Spanish translations: home-page-spanish-h1 1`] = `
<h1
  class="headline1"
>
  Esto es lo que escuchas
</h1>
`;

exports[`Home Page Language-Specific Snapshots renders home page with Spanish translations: home-page-spanish-h2 1`] = `
<h2
  class="headline2"
>
  Sube un enlace o canción (menos de 10MB y 10 minutos) para analizar su estado de ánimo, género, caso de uso, BPM, clave y más.
</h2>
`;
